### PR TITLE
cool-retro-term: fix on darwin

### PR DESCRIPTION
**Pull request by georgewhewell**
###### Motivation for this change

cool-retro-term had two problems on darwin:
 - binary not placed in `$PATH` due to macOS packaging
 - https://github.com/Swordfish90/cool-retro-term/issues/366

add a symlink to the macOS binary on darwin and update qmltermwidget makes it work

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/cool-retro-term/default.nix [hunk 1]
```diff
@@ -1,5 +1,5 @@
 { stdenv, fetchFromGitHub, qtbase, qtquick1, qmltermwidget,
-qtquickcontrols, qtgraphicaleffects, qmake }:
+qtquickcontrols, qtgraphicaleffects, qmake, makeWrapper, qtdeclarative }:
```
Comment (c0bw3b) [inline pkgs/applications/misc/cool-retro-term/default.nix]:
```suggestion
, qtquickcontrols, qtgraphicaleffects, qmake, makeWrapper, qtdeclarative }:
```
+ remove the trailing comma in the line above.
It's the usual way of presenting it in nixpkgs

File: pkgs/applications/misc/cool-retro-term/default.nix [hunk 2]
```diff
@@ -16,7 +16,7 @@ stdenv.mkDerivation rec {
     sed -i -e '/qmltermwidget/d' cool-retro-term.pro
   '';
 
-  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects ];
+  buildInputs = [ qtbase qtquick1 qmltermwidget qtquickcontrols qtgraphicaleffects qtdeclarative makeWrapper ];
```
Comment (c0bw3b) [inline pkgs/applications/misc/cool-retro-term/default.nix]: `makeWrapper` should be in `nativeBuildInputs` instead

File: pkgs/applications/misc/cool-retro-term/default.nix [hunk 3]
```diff
@@ -38,7 +46,7 @@ stdenv.mkDerivation rec {
     '';
     homepage = https://github.com/Swordifish90/cool-retro-term;
```
Comment (c0bw3b) [inline pkgs/applications/misc/cool-retro-term/default.nix]:
```suggestion
    homepage = https://github.com/Swordfish90/cool-retro-term;
```

File: pkgs/applications/misc/cool-retro-term/default.nix [hunk 4]
```diff
@@ -38,7 +46,7 @@ stdenv.mkDerivation rec {
...
     license = with stdenv.lib.licenses; [ gpl2 gpl3 ];
```
Comment (c0bw3b) [inline pkgs/applications/misc/cool-retro-term/default.nix]:
```suggestion
    license = stdenv.lib.licenses.gpl3Plus;
```

Source: https://github.com/Swordfish90/cool-retro-term/blob/master/app/qml/AboutDialog.qml#L97

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 5]
```diff
@@ -7,8 +7,8 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     repo = "qmltermwidget";
     owner = "Swordfish90";
-    rev = "v${version}";
-    sha256 = "0ca500mzcqglkj0i6km0z512y3a025dbm24605xyv18l6y0l2ny3";
+    rev = "48274c75660e28d44af7c195e79accdf1bd44963";
```
Comment (c0bw3b) [inline pkgs/development/libraries/qmltermwidget/default.nix]: Was #51572 not enough to fix Darwin build of `qmltermwidget` ?

Comment (georgewhewell) [reply pkgs/development/libraries/qmltermwidget/default.nix]: Unfortunately not- #51572 fixes the build, but there is another issue once it's running- https://github.com/Swordfish90/cool-retro-term/issues/366

File: pkgs/applications/misc/cool-retro-term/default.nix [hunk 6]
```diff
@@ -16,15 +16,23 @@ stdenv.mkDerivation rec {
...
     mv $out/usr/share $out/share
     mv $out/usr/bin $out/bin
     rmdir $out/usr
+  '' + stdenv.lib.optionalString stdenv.isDarwin ''
+    # Fixes "This application failed to start because it could not find or load the Qt
+    # platform plugin "cocoa"."
+    wrapProgram $out/bin/cool-retro-term.app/Contents/MacOS/cool-retro-term \
+     --set QT_QPA_PLATFORM_PLUGIN_PATH ${qtbase.bin}/lib/qt-*/plugins/platforms \
+     --set QT_PLUGIN_PATH ${qtbase.bin}/${qtbase.qtPluginPrefix} \
+     --set QML2_IMPORT_PATH ${qmltermwidget}/${qtbase.qtQmlPrefix}:${qtdeclarative.bin}/${qtbase.qtQmlPrefix}:${qtgraphicaleffects}/${qtbase.qtQmlPrefix}:${qtquickcontrols}/${qtbase.qtQmlPrefix}
```
Comment (Mic92) [inline pkgs/applications/misc/cool-retro-term/default.nix]: Is this also the case if you install retro-term in your nix profile?

Comment (georgewhewell) [reply pkgs/applications/misc/cool-retro-term/default.nix]: Oh, that works! Also works without wrapper using `nix run`. I have removed the wrapper, it seems like all that is needed is symlink to binary.

File: pkgs/development/libraries/qmltermwidget/default.nix (hunk 5 above)
Comment (alyssais) [inline pkgs/development/libraries/qmltermwidget/default.nix]: `version` should be changed to `2018-11-24` and name to `qmltermwidget-unstable-${version}` per https://nixos.org/nixpkgs/manual/#sec-package-naming.

Comment (georgewhewell) [reply pkgs/development/libraries/qmltermwidget/default.nix]: thx, done

**Pull request merged**